tests/phpunit/includes/test-wc-gateway-amazon-payments-advanced.php

```diff
@@ -136,4 +123,118 @@ public function test_classic_process_payment_succeeds() : void {
 			$mock_gateway->process_payment( $order->get_id() )
 		);
 	}
+
+	/**
+	 * Test maybe_separator_and_checkout_button_single_product method.
+	 *
+	 * @return void
+	 */
+	public function test_maybe_separator_and_checkout_button_single_product() : void {
```

Comment:
The test case is a bit complicated. IIUC, we can split it 2:
- one test case covering non-product context
- one test with a product, but also a data provider providing stock statuses

I might miss something, but each test case covers one scenario.

Reply:
I wish I had more capacity to spend here, but we are good at having the tests.
My comments imply a bit of pair programming, to improve the test structure a bit - let's ignore my comment.